.meta/sources/kubernetes_logs.toml.erb

```diff
@@ -0,0 +1,103 @@
+[sources.kubernetes_logs]
+title = "Kubernetes Logs"
+noun = "Kubernetes Logs"
+beta = false
+common = true
+delivery_guarantee = "best_effort"
+description = "Deploy Vector into your Kubernetes cluster and collect logs from Pods."
+features = [
+  "Collect logs from Pods when running in Kubernetes.",
+  "Enrich your logs with Pod metadata.",
+  "Exclude logs from collecion via annoting Pods with `vector.dev/exclude: \"true\"`.",
+  "Automatically merge partial messages into one event.",
+  "Checkpoint your position to ensure data is not lost between restarts.",
+]
+function_category = "collect"
+output_types = ["log"]
+requirements.kubernetes = ">= 1.14" # TODO: Link to MSKV here. MSKV is shared accross all k8s components.
+# requirements.kubernetes_in_cluster = true # TODO: Denote that in-cluster execution is required.
+strategies = ["daemon"]
+through_description = "Kubernetes logs from Pods"
+
+[sources.kubernetes_logs.env_vars.KUBERNETES_SERVICE_HOST]
+type = "string"
+common = false
+required = true
+examples = ["10.96.0.1"]
+relevant_when = {requirements.kubernetes_in_cluster = true}
+description = """\
+An in-cluster Kubernetes API `Service` host. \
+Provided automatically by the Kubernetes when running in the cluster.\
+"""
+
+[sources.kubernetes_logs.env_vars.KUBERNETES_SERVICE_PORT]
+type = "string"
+common = false
+required = true
+examples = ["443"]
+relevant_when = {requirements.kubernetes_in_cluster = true}
+description = """\
+An in-cluster Kubernetes API `Service` port. \
+Provided automatically by the Kubernetes when running in the cluster.\
+"""
+
+<%= render("_partials/fields/_component_options.toml", type: "source", name: "kubernetes_logs") %>
+
+[sources.kubernetes_logs.options.self_node_name]
+type = "string"
+common = false
+default = "${VECTOR_SELF_NODE_NAME}"
+description = """\
+The name of the Kubernetes `Node` this Vector instance runs at. Configured to \
+use an env var by default, to be evaluated to a value provided by Kubernetes \
+at Pod deploy time.\
+"""
+
+[sources.kubernetes_logs.options.auto_partial_merge]
+type = "bool"
+common = false
+default = true
+description = """\
+Automatically merge partial messages into a signle event. \
+Partial here is in respect to messages that were split by the Kubernetes \
+Container Runtime log driver.\
+"""
+
+[sources.kubernetes_logs.options.annotation_fields]
```

Comment:
@MOZGIII I like that you nested field configuration under a key, but it deviates from other sources. For example, the `file` source uses root-level options like `host_key` and `file_key`. I don't think we should change it right now but wanted you to be aware. I think https://github.com/timberio/vector/issues/3463 is much better solution to all of this.